### ISSUE TITLE
Phase 1 / v1.x release

### ISSUE THREAD

**Issue by springmeyer** [mapbox/vector-tile#2]
I propose the v1.x release focus exclusively on a common set of headers with enumerations for Mapbox Vector Tile types. See #1 for how this could reduce duplication: e.g. between https://github.com/mapbox/mapnik-vector-tile/blob/master/src/vector_tile_config.hpp and https://github.com/Project-OSRM/osrm-backend/blob/master/include/util/vector_tile.hpp.

Further phases (2.x and 3.x) would focus on adding decoding and encoding support.

**Comment by mapsam** [comment]
@springmeyer @flippmoke which syntax would you prefer?

### constant expression

```cpp
const constexpr std::uint32_t LAYER_TAG = 3;
```

### enum namespace

```cpp
enum Tile_Encoding : std::uint32_t
{
    LAYERS = 3
};
```

**Comment by mapsam** [comment]
Either way, how do we want to access the constants? We can be hyper explicit in the namespaces or just have command underscore `_` nomenclature.

Explicit namespace
```
writer.add_uint32(vector_tile::Encoding::Tile::VERSION, 2);
```

Underscores
```
writer.add_uint32(vector_tile::VERSION_TAG, 2);
```

Or something in between?

**Comment by springmeyer** [comment]
Great work @mapsam, closed by #4
